src/sortStudents.ts

```diff
@@ -1,16 +1,73 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  sortOrder: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+  const sortDirection = (sortOrder === 'asc') ? 1 : -1;
+
+  switch (sortBy) {
+    case SortType.Name:
+
+      sortedStudents.sort(
+        // eslint-disable-next-line max-len
+        (firstStudent: Student, secondStudent: Student) => firstStudent.name.localeCompare(secondStudent.name),
+      );
+      break;
+
+    case SortType.Surname:
+      return sortedStudents.sort(
+        // eslint-disable-next-line max-len
+        (firstStudent: Student, secondStudent: Student) => firstStudent.surname.localeCompare(secondStudent.surname),
+      );
+
+    case SortType.Age:
+      return sortedStudents.sort(
+        // eslint-disable-next-line max-len
+        (firstStudent: Student, secondStudent: Student) => sortDirection * (firstStudent.age - secondStudent.age),
+      );
+
+    case SortType.Married:
+      sortedStudents.sort((firstStudent: Student, secondStudent: Student) => (
+        // eslint-disable-next-line max-len
+        firstStudent.married > secondStudent.married ? sortDirection : sortDirection * -1
+      ));
+      break;
+
+    case SortType.AverageGrade:
+      sortedStudents.sort((firstStudent: Student, secondStudent: Student) => {
+        // eslint-disable-next-line max-len
+        const avgFirstStudent = firstStudent.grades.reduce((acc, curr) => acc + curr, 0)
+          / firstStudent.grades.length;
+        // eslint-disable-next-line max-len
+        const avgSecondStudent = secondStudent.grades.reduce((acc, curr) => acc + curr, 0)
+          / secondStudent.grades.length;
```

Comment:
```suggestion
        const avgSecondStudent = secondStudent.grades.reduce(
          (acc, curr) => acc + curr, 0,
        ) / secondStudent.grades.length;
```
do not use `eslint-disable-next-line max-len` remove it all over your code and format the code